packages/internal/src/upgrade-api.js

```diff
@@ -45,5 +45,23 @@ harden(makeUpgradeDisconnection);
  * @returns {reason is UpgradeDisconnection}
  */
 export const isUpgradeDisconnection = reason =>
-  isFrozen(reason) && matches(reason, UpgradeDisconnectionShape);
+  reason != null && // eslint-disable-line eqeqeq
+  isFrozen(reason) &&
+  matches(reason, UpgradeDisconnectionShape);
 harden(isUpgradeDisconnection);
+
+/**
+ * Returns whether a reason is a 'vat terminated' error generated when an object
```

Comment:
I understand this as an expedient measure, until we distinguish this rejection condition some other way. Either, like UpgradeDisconnection, use something other than an Error as a rejection to be tested. For an error, the only content that a program should make a semantic decision on is `.name`. Most emphatically, code should not make a semantic decision on an error `.message`.

So, an encapsulated temporary expedient measure is fine if clearly marked as such, and with a filed issue to fix it.

Reply:
( @mhofman let me know the relevant issue is https://github.com/Agoric/agoric-sdk/issues/9582 )

Please add a comment here with that link.